test/package.json

```diff
@@ -0,0 +1,30 @@
+{
+  "name": "@mui-x-internal/test",
```

Comment:
Had to create the package to resolve `eslint` issues after switching to pnpm.
Feel free to suggest a better name as we could end up replacing the `test/*` import with this package name and ditch the explicit `test/*` alias.
I've gone with something similar to the MUI Core internal package, where they start with `@mui-internal`.
WDYT about `@mui-internal/x-test? 🤔